components/authz-service/storage/v2/postgres/postgres_test.go

```diff
@@ -3791,6 +3791,103 @@ func TestListRulesForProject(t *testing.T) {
+			ctx := context.Background()
+			projID := "project-1"
+			insertTestProject(t, db, projID, "first project", storage.Custom)
+
+			condition, err := storage.NewCondition(storage.Node,
+				[]string{"chef-server-1"}, storage.ChefServer, storage.MemberOf)
+			require.NoError(t, err)
+			rule, err := storage.NewRule("first-rule", projID, "the very first rule", condition.Type,
+				[]storage.Condition{condition})
+			require.NoError(t, err)
+			insertAppliedRule(t, db, &rule)
+
+			updatedCondition, err := storage.NewCondition(storage.Node,
+				[]string{"new-chef-server"}, storage.ChefServer, storage.Equals)
+			require.NoError(t, err)
+			updatedRule, err := storage.NewRule(rule.ID, projID, "updated rule name", condition.Type,
+				[]storage.Condition{updatedCondition})
+			insertStagedRule(t, db, &updatedRule, false)
+
+			resp, err := store.ListRulesForProject(ctx, projID)
+			assert.NoError(t, err)
+			assert.ElementsMatch(t, []*storage.Rule{&updatedRule}, resp)
+		}},
+		{"when there are staged changes for some of the project's rules, returns mix of staged and applied rules", func(t *testing.T) {
+			ctx := context.Background()
+			projID := "project-1"
+			insertTestProject(t, db, projID, "first project", storage.Custom)
+
+			condition, err := storage.NewCondition(storage.Node,
+				[]string{"chef-server-1"}, storage.ChefServer, storage.MemberOf)
+			require.NoError(t, err)
+			rule, err := storage.NewRule("first-rule", projID, "the very first rule", condition.Type,
+				[]storage.Condition{condition})
+			require.NoError(t, err)
+			insertAppliedRule(t, db, &rule)
+
+			updatedCondition, err := storage.NewCondition(storage.Node,
+				[]string{"new-chef-server"}, storage.ChefServer, storage.Equals)
+			require.NoError(t, err)
+			updatedRule, err := storage.NewRule(rule.ID, projID, "updated rule name", updatedCondition.Type,
+				[]storage.Condition{updatedCondition})
+			require.NoError(t, err)
+			insertStagedRule(t, db, &updatedRule, false)
+
+			appliedRule := insertAppliedRuleWithMultipleConditions(t, db, projID, condition.Type)
+
+			resp, err := store.ListRulesForProject(ctx, projID)
+			assert.NoError(t, err)
+			assert.ElementsMatch(t, []*storage.Rule{&updatedRule, appliedRule}, resp)
+		}},
+		{"when a project's rule has been marked for deletion, it will not be returned", func(t *testing.T) {
+			ctx := context.Background()
+			projID := "foo-project"
+			insertTestProject(t, db, projID, "first project", storage.Custom)
+
+			condition, err := storage.NewCondition(storage.Node,
+				[]string{"chef-server-1"}, storage.ChefServer, storage.MemberOf)
+			require.NoError(t, err)
+			rule, err := storage.NewRule("foo-rule", projID, "coo foo rule", condition.Type, []storage.Condition{condition})
+			assert.NoError(t, err)
+			insertAppliedRule(t, db, &rule)
+
+			// mark for deletion
+			insertStagedRule(t, db, &rule, true)
```

Comment:
hmm maybe we should add another helper `insertDeletedStagedRule` instead of needing the comment 🤔